### PR TITLE
Feature/added teams webhook

### PR DESCRIPTION
**Pull request by Iamprashanth-1**
# Summary

Added Team Webhook URL to Mage-AI so If we configure Teams webhook URL the pipeline status will be sent to respective teams channel.

# Tests

Tested by adding Teams webhook URL in the metadata.YAML file

cc:

### CODE REVIEW

**Comment by tommydangerous** [comment]
👀

File: mage_ai/orchestration/notification/config.py [hunk 1]
```diff
@@ -10,10 +11,12 @@
 class NotificationConfig(BaseConfig):
     email_config: EmailConfig = None
     slack_config: SlackConfig = None
+    teams_config: TeamsConfig = None
 
     @classmethod
     def load(self, config_path: str = None, config: Dict = None):
         notification_config = super().load(config_path=config_path, config=config)
+        print(notification_config ,'hell')
```
Comment (tommydangerous) [inline mage_ai/orchestration/notification/config.py:19]: Can we remove this please?

File: mage_ai/orchestration/notification/config.py [hunk 2]
```diff
@@ -23,6 +26,16 @@ def load(self, config_path: str = None, config: Dict = None):
             except Exception:
                 traceback.print_exc()
                 notification_config.slack_config = None
+        if notification_config.teams_config is not None and \
+                type(notification_config.teams_config) is dict:
+            try:
+                notification_config.teams_config = TeamsConfig.load(
+                    config=notification_config.teams_config,
+                )
+                print(notification_config.teams_config ,'hello')
```
Comment (tommydangerous) [inline mage_ai/orchestration/notification/config.py:35]: Please remove print

File: mage_ai/orchestration/notification/sender.py [hunk 3]
```diff
@@ -18,6 +19,11 @@ def send(
             return
         if self.config.slack_config is not None and self.config.slack_config.is_valid:
             send_slack_message(self.config.slack_config, message)
+
+        if self.config.teams_config is not None and self.config.teams_config.is_valid:
+            send_teams_message(self.config.teams_config, message)
+            
+
```
Comment (tommydangerous) [inline mage_ai/orchestration/notification/sender.py:26]: Remove extra line please.

File: mage_ai/orchestration/notification/config.py (hunk 1 above)
Comment (Iamprashanth-1) [reply mage_ai/orchestration/notification/config.py:19]: sure

Comment (Iamprashanth-1) [reply mage_ai/orchestration/notification/config.py]: Done.

File: mage_ai/orchestration/notification/sender.py (hunk 3 above)
Comment (Iamprashanth-1) [reply mage_ai/orchestration/notification/sender.py]: Done

File: mage_ai/services/teams/teams.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from mage_ai.services.teams.config import TeamsConfig
+import json
+import requests
+
+
+def send_teams_message(config: TeamsConfig, message: str ,title = "Mage Ai Pipeline Run Status Logs",) -> None:
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:6]: Can we change the default message to: Mage pipeline run status logs.

Also, please use single quotes

File: mage_ai/services/teams/teams.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    requests.post(
+        url = config.webhook_url,
+        json = {
+            "summary": title,
+            "sections": [{
+                "activityTitle": title,
+                "activitySubtitle": message
+            }],
+
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:15]: Remove empty line

File: mage_ai/services/teams/teams.py (hunk 4 above)
Comment (Iamprashanth-1) [reply mage_ai/services/teams/teams.py:6]: Ok Sure

File: mage_ai/services/teams/teams.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+from mage_ai.services.teams.config import TeamsConfig
+import json
+import requests
+
+
+def send_teams_message(config: TeamsConfig, message: str ,title = "Mage Ai Pipeline Run Status Logs",) -> None:
+    requests.post(
+        url = config.webhook_url,
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:8]: No spaces around =

File: mage_ai/services/teams/teams.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+        json = {
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:9]: No spaces around =

File: mage_ai/services/teams/teams.py (hunk 4 above)
Comment (Iamprashanth-1) [reply mage_ai/services/teams/teams.py]: Done

File: mage_ai/services/teams/teams.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
+from mage_ai.services.teams.config import TeamsConfig
+import json
+import requests
+
+
+def send_teams_message(config: TeamsConfig, message: str ,title = 'Mage pipeline run status logs',) -> None:
+    requests.post(
+        url = config.webhook_url,
+        json = {
+            "summary": title,
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:10]: Please use single quotes.

File: mage_ai/services/teams/teams.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+            "sections": [{
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:11]: Please use single quotes.

File: mage_ai/services/teams/teams.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+                "activityTitle": title,
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:12]: Please use single quotes.

File: mage_ai/services/teams/teams.py [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+                "activitySubtitle": message
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:13]: Please use single quotes.

File: mage_ai/tests/orchestration/notification/constants.py [hunk 12]
```diff
@@ -13,3 +13,10 @@
         webhook_url='test_webhook_url',
     )
 )
+
+
+TEAMS_NOTIFICATION_CONFIG = dict(
+    teams_config=dict(
+        webhook_url='test_webhook_url'
```
Comment (tommydangerous) [inline mage_ai/tests/orchestration/notification/constants.py:20]: Please add trailing comma

File: mage_ai/tests/orchestration/notification/constants.py [hunk 13]
```diff
@@ -13,3 +13,10 @@
...
+    )
```
Comment (tommydangerous) [inline mage_ai/tests/orchestration/notification/constants.py:21]: Please add trailing comma

File: mage_ai/tests/orchestration/notification/test_config.py [hunk 14]
```diff
@@ -3,6 +3,7 @@
 from mage_ai.tests.orchestration.notification.constants import (
     EMAIL_NOTIFICATION_CONFIG,
     SLACK_NOTIFICATION_CONFIG,
+    TEAMS_NOTIFICATION_CONFIG
```
Comment (tommydangerous) [inline mage_ai/tests/orchestration/notification/test_config.py:6]: Please add trailing comma

File: mage_ai/tests/orchestration/notification/test_sender.py [hunk 15]
```diff
@@ -5,6 +5,8 @@
 from mage_ai.tests.orchestration.notification.constants import (
     EMAIL_NOTIFICATION_CONFIG,
     SLACK_NOTIFICATION_CONFIG,
+    TEAMS_NOTIFICATION_CONFIG
```
Comment (tommydangerous) [inline mage_ai/tests/orchestration/notification/test_sender.py:8]: Please add trailing comma

File: mage_ai/tests/orchestration/notification/test_sender.py [hunk 16]
```diff
@@ -5,6 +5,8 @@
...
+
```
Comment (tommydangerous) [inline mage_ai/tests/orchestration/notification/test_sender.py:9]: Remove extra line

File: mage_ai/services/teams/teams.py (hunk 7 above)
Comment (Iamprashanth-1) [reply mage_ai/services/teams/teams.py:9]: there are spaces right before and after

File: mage_ai/tests/orchestration/notification/test_sender.py (hunk 16 above)
Comment (Iamprashanth-1) [reply mage_ai/tests/orchestration/notification/test_sender.py]: Done with changes

File: mage_ai/services/teams/teams.py (hunk 7 above)
Comment (tommydangerous) [reply mage_ai/services/teams/teams.py:9]: Can you change it to json={

File: mage_ai/services/teams/teams.py [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+def send_teams_message(config: TeamsConfig, message: str ,title = 'Mage pipeline run status logs',) -> None:
+    requests.post(
+        url = config.webhook_url,
+        json = {
+            'summary': title,
+            'sections': [{
+                'activityTitle': title,
+                'activitySubtitle': message
+            }],
+        }
```
Comment (tommydangerous) [inline mage_ai/services/teams/teams.py:15]: Please add trailing comma

File: mage_ai/tests/orchestration/notification/constants.py (hunk 13 above)
Comment (tommydangerous) [reply mage_ai/tests/orchestration/notification/constants.py:21]: Please add trailing comma

File: mage_ai/services/teams/teams.py (hunk 17 above)
Comment (Iamprashanth-1) [reply mage_ai/services/teams/teams.py:15]: Done

File: mage_ai/tests/orchestration/notification/constants.py (hunk 13 above)
Comment (Iamprashanth-1) [reply mage_ai/tests/orchestration/notification/constants.py:21]: Done

**Review by tommydangerous — APPROVED** [review]
LGTM

**Pull request merged**